components/dropdown/dropdown-menu.js

```diff
@@ -268,10 +268,9 @@ if (usePopoverMixin) {
 		}
 
 		__getMenuElement() {
-			if (!this.shadowRoot) return undefined;
-			return this.shadowRoot.querySelector('.d2l-dropdown-content-slot')
-				.assignedNodes().filter(node => node.hasAttribute
-					&& (node.getAttribute('role') === 'menu' || node.getAttribute('role') === 'listbox'))[0];
+			return this.shadowRoot?.querySelector('.d2l-dropdown-content-slot')
+				.assignedElements({ flatten: true })
```

Comment:
The `flatten: true` is required in order for `d2l-dropdown-menu` to locate the `d2l-menu` in the consumer's slot, such as the slot provided by `d2l-button-split`. Otherwise, it would just find the `d2l-button-split`'s menu slot instead of the stuff inside it.
```html
<d2l-button-split>
  --shadowDOM
    <d2l-dropdown-menu>
      <slot name="menu"></slot>
    </d2l-dropdown-menu>

  <d2l-menu slot="menu">
    ...
  </d2l-menu>
</d2l-button-split>
```